### PR TITLE
Datastore: id from #3832 pull request with unit test

### PR DESCRIPTION
**Pull request by chemelnucfin**
See #3832.  Just following up with it.

I'm going to leave the document in place as it is the same as the other function.  I can make a separate pull request to remove implementation details.

### CODE REVIEW

File: datastore/tests/unit/test_entity.py [hunk 1]
```diff
@@ -36,6 +36,7 @@ def test_ctor_defaults(self):
         entity = klass()
         self.assertIsNone(entity.key)
         self.assertIsNone(entity.kind)
+        self.assertIsNone(entity.id)
```
Comment (dhermes) [inline datastore/tests/unit/test_entity.py]: @chemelnucfin IMO it's "bad form" to check a `@property` in the constructor test. I view these assertions as essentially checking whats in `entity.__dict__`

File: datastore/tests/unit/test_entity.py [hunk 2]
```diff
@@ -192,6 +193,13 @@ def test__eq__same_value_different_meanings(self):
 
         self.assertFalse(entity1 == entity2)
 
+    def test_id(self):
+        from google.cloud.datastore.key import Key
+
+        key = Key(_KIND, _ID, project=_PROJECT)
+        entity = self._make_one(key=key)
+        self.assertTrue(entity.id == _ID)
```
Comment (dhermes) [inline datastore/tests/unit/test_entity.py]: Please use `assertEqual`

File: datastore/tests/unit/test_entity.py [hunk 3]
```diff
@@ -192,6 +193,13 @@ def test__eq__same_value_different_meanings(self):
 
         self.assertFalse(entity1 == entity2)
 
+    def test_id(self):
```
Comment (dhermes) [inline datastore/tests/unit/test_entity.py]: You also need a test for the case when it is `None`

File: datastore/google/cloud/datastore/entity.py [hunk 4]
```diff
@@ -183,6 +183,19 @@ def kind(self):
...
+    def id(self):
+        """Get the id of the current entity.
+
+        .. note::
+          This relies entirely on the :class:`gcloud.datastore.key.Key`
+          set on the entity.  That means that we're not storing the id
+          of the entity at all, just the properties and a pointer to a
+          Key which knows its Id.
+        """
+        if self.key is not None:
```
Comment (dhermes) [inline datastore/google/cloud/datastore/entity.py]: From [PEP8][1]

> Be consistent in return statements. Either all return statements in a function should return an expression, or none of them should.

So this should be:

```python
if self.key is None:
    return None
else:
    return self.key.id
```

[1]: https://www.python.org/dev/peps/pep-0008/#programming-recommendations

File: datastore/google/cloud/datastore/entity.py [hunk 5]
```diff
@@ -183,6 +183,19 @@ def kind(self):
         if self.key:
             return self.key.kind
 
+    @property
+    def id(self):
+        """Get the id of the current entity.
+
+        .. note::
+          This relies entirely on the :class:`gcloud.datastore.key.Key`
```
Comment (dhermes) [inline datastore/google/cloud/datastore/entity.py]: 1. There is no "Sphinx / rST lint" but I've seen @lukesneeringer use 4 space indents after a `.. note::` and IMO it's good form to leave an empty line between the ``.. note::` directive and the actual content.
1. There is no such thing a `gcloud` any more, not sure where this text comes from.
1. Please use ID (not "id")

File: datastore/tests/unit/test_entity.py (hunk 1 above)
Comment (chemelnucfin) [reply datastore/tests/unit/test_entity.py]: I was just following what was there previously.

File: datastore/tests/unit/test_entity.py (hunk 3 above)
Comment (chemelnucfin) [reply datastore/tests/unit/test_entity.py]: It was in the constructor test.  I can add one I guess.

File: datastore/tests/unit/test_entity.py (hunk 1 above)
Comment (dhermes) [reply datastore/tests/unit/test_entity.py]: Great. Does my approach seem logical?

File: datastore/tests/unit/test_entity.py (hunk 3 above)
Comment (dhermes) [reply datastore/tests/unit/test_entity.py]: Please do.

File: datastore/google/cloud/datastore/entity.py (hunk 5 above)
Comment (chemelnucfin) [reply datastore/google/cloud/datastore/entity.py]: I will follow up with the documentation changes for both on a separate PR.  It would be inconsistent to change one and not the other and it would detract from this PR if I sent it together.

Comment (dhermes) [reply datastore/google/cloud/datastore/entity.py]: Just change this part and feel free to send a follow-up PR to update the other part.

Comment (chemelnucfin) [reply datastore/google/cloud/datastore/entity.py]: See #4641.  I changed them both there.
